hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -401,12 +401,12 @@ private List<CleanFileInfo> getReplacedFilesEligibleToClean(List<String> savepoi
   }
 
   /**
-   * Gets the latest version < instantTime. This version file could still be used by queries.
+   * Gets the latest version <= instantTime. This version file could still be used by queries.
    */
   private String getLatestVersionBeforeCommit(List<FileSlice> fileSliceList, HoodieInstant instantTime) {
     for (FileSlice file : fileSliceList) {
       String fileCommitTime = file.getBaseInstantTime();
```

Comment:
The method name indicates `before` ?

Reply:
Year, it's before.
Double check with you, if we set CLEANER_COMMITS_RETAINED=n, do we actualy retain n+1 commits by design? For example, if we have a fileGroup file_001.parquet,file_002.parquet,file_003.parquet,file_004.parquet, we set CLEANER_COMMITS_RETAINED=2, file_002.parquet,file_003.parquet,file_004.parquet will stay after a clean.